python/lsst/ts/attcpip/csc.py

```diff
@@ -225,35 +234,53 @@ async def start_clients(self) -> None:
             cmd_evt_port = self.simulator.cmd_evt_server.port
             telemetry_port = self.simulator.telemetry_server.port
 
-        self.cmd_evt_client = tcpip.Client(
-            host=host, port=cmd_evt_port, log=self.log, name="CmdEvtClient"
-        )
-        self.telemetry_client = tcpip.Client(
-            host=host, port=telemetry_port, log=self.log, name="TelemetryClient"
-        )
+        # Stop the clients in case the configuration has changed.
+        if self.cmd_evt_client.connected and (
+            self.cmd_evt_client.host != host or self.cmd_evt_client.port != cmd_evt_port
+        ):
+            await self._stop_cmd_evt_task_and_client()
+        if self.telemetry_client.connected and (
+            self.telemetry_client.host != host
+            or self.telemetry_client.port != telemetry_port
+        ):
+            await self._stop_telemetry_task_and_client()
+
+        if not self.cmd_evt_client.connected:
+            self.log.debug("Starting cmd_evt client.")
+            self.cmd_evt_client = tcpip.Client(
+                host=host, port=cmd_evt_port, log=self.log, name="CmdEvtClient"
+            )
+            await self.cmd_evt_client.start_task
+            self._event_task = asyncio.create_task(self.cmd_evt_loop())
+
+        if not self.telemetry_client.connected:
+            self.log.debug("Starting telemetry client.")
+            self.telemetry_client = tcpip.Client(
+                host=host, port=telemetry_port, log=self.log, name="TelemetryClient"
+            )
+            await self.telemetry_client.start_task
+            self._telemetry_task = asyncio.create_task(self.telemetry_loop())
```

Comment:
As the comments above, if you lost one client for the internet issue, the other one should fail as well.

Reply:
This is tasking care of more situations than only when the network connection is lost. For instance, when going to STANDBY, the telemetry needs to be stopped. That situation is captured by this code as well as losing the network connection.